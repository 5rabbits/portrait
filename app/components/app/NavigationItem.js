import React, { PureComponent } from 'react';
import { Link } from 'react-router';
import classNames from 'classnames';

export default class Navigation extends PureComponent {
  render() {
    <li clasName={classNames({ "active": this.props.isActive })}>
      <Link to={this.props.href}>
        <i className={`fa mr-xs ${this.props.icon}`}></i>
        <span className="hidden-sm">{this.props.label}</span>
      </Link>
    </li>
  }
}
