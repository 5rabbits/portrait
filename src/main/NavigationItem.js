import React from 'react'
import classNames from 'classnames'

const NavigationItem = props => {
  const { component: Component, isActive, icon, label, ...other } = props

  debugger
  return (
    <li className={classNames({ active: isActive })}>
      <Component {...other}>
        <i className={`fa mr-xs ${icon}`} />
        <span className="hidden-sm">{label}</span>
      </Component>
    </li>
  )
}

NavigationItem.defaultProps = {
  component: 'a',
}

export default NavigationItem
