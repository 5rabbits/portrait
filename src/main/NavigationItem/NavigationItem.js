import React, { PropTypes } from 'react'
import cx from 'classnames'
import FixedWidth from 'helpers/FixedWidth'

export default class NavigationItem extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    containerProps: PropTypes.object,
    component: PropTypes.oneOfType([
      PropTypes.string, PropTypes.func,
    ]),
    isActive: PropTypes.bool,
    icon: PropTypes.node,
    label: PropTypes.node,
  }

  static defaultProps = {
    component: 'a',
    containerProps: {},
  }

  static contextTypes = {
    isNestedNavigationItem: PropTypes.bool,
    hideNavigationItemText: PropTypes.bool,
  }

  static childContextTypes = {
    isNestedNavigationItem: PropTypes.bool,
  }

  state = {
    open: false,
  }

  getChildContext() {
    return {
      isNestedNavigationItem: true,
    }
  }

  toggleDropdown = () => {
    this.setState({ open: !this.state.open })
  }

  render() {
    const {
      component: Component, isActive, icon, label, children, containerProps,
      ...other
    } = this.props

    const { open } = this.state
    const { isNestedNavigationItem, hideNavigationItemText } = this.context
    const isNested = isNestedNavigationItem || false
    const isDropdown = children != null
    const withText = !hideNavigationItemText || isNested

    return (
      <div
        {...containerProps}
        className={cx('navigation-item', containerProps.className, {
          'navigation-item--active': isActive,
          'navigation-item--top-level': !isNested,
          'navigation-item--with-text': withText,
          'navigation-item--without-text': !withText,
          'navigation-item--open': open,
          dropdown: isDropdown,
        })}
        >
        <Component
          {...other}
          onClick={isDropdown ? this.toggleDropdown : undefined}
          >
          {icon &&
            <span className="navigation-item__icon">
              {icon}
            </span>
          }

          <FixedWidth>
            <span className="navigation-item__label">
              {label}
            </span>
          </FixedWidth>

          {isDropdown &&
            <span className="caret navigation-item__caret" />
          }
        </Component>

        {isDropdown &&
          <div
            className={cx('navigation-item__dropdown', {
              'navigation-item__dropdown--visible': open,
            })}
            >
            {children}
          </div>
        }
      </div>
    )
  }
}
