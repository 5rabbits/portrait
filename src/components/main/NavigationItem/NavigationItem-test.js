import { React, toJSON, mount } from 'test/helper'
import NavigationItem from 'main/NavigationItem'

describe('NavigationItem', () => {
  it('should render with minimum props', () => {
    const component = mount(<NavigationItem />)

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render nested NavigationItem as a dropdown', () => {
    const component = mount(
      <NavigationItem>
        <NavigationItem />
        <NavigationItem />
      </NavigationItem>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should allow to define an icon', () => {
    const component = mount(<NavigationItem icon="test" />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
