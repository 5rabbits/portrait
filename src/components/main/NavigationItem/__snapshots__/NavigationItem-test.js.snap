// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem should allow to define an icon 1`] = `
<NavigationItem
  component="a"
  containerProps={Object {}}
  icon="test"
>
  <li
    className="navigation-item--top-level"
  >
    <a>
      <span
        className="mr-xs navigation-item__icon"
      >
        test
      </span>
      <span
        className="hidden-sm"
      />
    </a>
  </li>
</NavigationItem>
`;

exports[`NavigationItem should render nested NavigationItem as a dropdown 1`] = `
<NavigationItem
  component="a"
  containerProps={Object {}}
>
  <li
    className="dropdown navigation-item--top-level"
  >
    <a
      data-toggle="dropdown"
    >
      <span
        className="hidden-sm"
      />
      <span
        className="caret navigation-item__caret"
      />
    </a>
    <ul
      className="dropdown-menu"
    >
      <NavigationItem
        component="a"
        containerProps={Object {}}
      >
        <li
          className=""
        >
          <a>
            <span
              className=""
            />
          </a>
        </li>
      </NavigationItem>
      <NavigationItem
        component="a"
        containerProps={Object {}}
      >
        <li
          className=""
        >
          <a>
            <span
              className=""
            />
          </a>
        </li>
      </NavigationItem>
    </ul>
  </li>
</NavigationItem>
`;

exports[`NavigationItem should render with minimum props 1`] = `
<NavigationItem
  component="a"
  containerProps={Object {}}
>
  <li
    className="navigation-item--top-level"
  >
    <a>
      <span
        className="hidden-sm"
      />
    </a>
  </li>
</NavigationItem>
`;
